library/std/src/f32.rs

```diff
@@ -1154,4 +1154,54 @@ impl f32 {
         let x = unsafe { cmath::lgammaf_r(self, &mut signgamp) };
         (x, signgamp)
     }
+
+    /// Float addition that allows optimizations based on algebraic rules.
+    #[rustc_allow_incoherent_impl]
```

Comment:
These methods should be in `core` so that they don't need this, and are also usable in `no_std` programs.

Reply:
Updated, thanks!